lib/super_digit.rb

```diff
@@ -1,15 +1,34 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) where n is the amount of times the sum is greater than 10 
+# Space Complexity - O(n) where n is the number of digits in n
 def super_digit(n)
-   
+  return n if n < 10
+
+  sum = 0 
+
+  while n > 0
+    digit = n % 10
+    n /= 10
+    sum += digit
+  end
+
+  super_digit(sum)
+
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def refined_super_digit(n, k)
-    
+  refined = ""
+
+  k.times do 
+    refined += n.to_s 
+  end 
+  refined = refined.to_i
+
+  super_digit(refined)
+
```

Comment:
Consider a dynamic programming approach:

```suggestion
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```